spec/GraphQLOverHTTP.md

```diff
@@ -178,49 +227,74 @@ A request for execution should contain the following request parameters:
+["request"](https://spec.graphql.org/draft/#request) in the GraphQL
+specification you should note the {GraphQL schema} and "initial value" are not
+included in the GraphQL-over-HTTP {request}, they are handled by the {server}
+based on the URL used.
+
+Note: Be aware that `query` is a misleading parameter name as its value is a
+string describing one or more operations, each of which may be a query, mutation
+or subscription. A better name would have been `document`, but the term `query`
+is well established.
 
-Note: depending on the serialization format used, values of the aforementioned
+Note: Depending on the serialization format used, values of the aforementioned
 parameters can be encoded differently but their names and semantics must stay
 the same.
 
-Note: specifying `null` in JSON (or equivalent values in other formats) as
+Note: Specifying `null` in JSON (or equivalent values in other formats) as
 values for optional request parameters is equivalent to not specifying them at
 all.
 
-Note: {variables} and {extensions}, if set, must have a map as its value.
+Note: {variables} and {extensions}, if set, must have a map as their value.
+
+## Accept
+
+A client SHOULD indicate the media types that it supports in responses using the
+`Accept` HTTP header as specified in
+[RFC7231](https://datatracker.ietf.org/doc/html/rfc7231).
+
+If the client supplies an `Accept` header, the client SHOULD include the media
+type `application/graphql+json` in the `Accept` header.
+
+### Legacy Watershed
+
+Before `2025-01-01T00:00:00Z`, if the client supplies an `Accept` header, the
+header SHOULD include the `application/json` media type. After this watershed,
+this is no longer necessary.
+
+It is RECOMMENDED that a client set the `Accept` header to
+`application/graphql+json; charset=utf-8, application/json; charset=utf-8`.
+
+Note: This recommended header enables compatibility with legacy servers whilst
+still leveraging modern features if available in the server.
 
 ## GET
 
-For HTTP GET requests, the query parameters MUST be provided in the query
-component of the request URL in the form of `key=value` pairs with `&` symbol as
-a separator and both the key and value should have their "reserved" characters
-percent-encoded as specified in
-[section 2 of RFC3986](https://tools.ietf.org/html/rfc3986#section-2). The
-`variables` parameter, if used, MUST be represented as a URL-encoded JSON
+For HTTP GET requests, the GraphQL request parameters MUST be provided in the
+query component of the request URL in the `application/x-www-form-urlencoded`
+format as specified by
+[WhatWG's URLSearchParams class](https://url.spec.whatwg.org/#interface-urlsearchparams).
+
+The `query` parameter MUST be the string representation of the Source Text of
+the Document as specified in
+[the Language section of the GraphQL specification](https://spec.graphql.org/draft/#sec-Language).
+
+The `variables` parameter, if used, MUST be represented as a URL-encoded JSON
```

Comment:
Why not specify the same thing for `extensions`?

Reply:
I've never noticed extensions being used in a URL, but you noted it is used for APQ so I'll add it :+1: